### ISSUE TITLE
Items do not appear if the client or server does not have the mod

### ISSUE THREAD

**Issue by OneDefauter** [Crystal-Nest/soul-fire-d#8]
CurseForge comment
https://www.curseforge.com/minecraft/mc-mods/soul-fired?comment=14

Vanilla (withou mod)
https://youtu.be/dlMtqTkAeEA

Fabric (with mod)
https://youtu.be/MmcXnd_dqlE

**Comment by Crystal-Spider** [comment]
So, correct me if I'm wrong, but it seems to me that Torch hit! alone causes drops to be invisible, while Torch hit! + Soul Fire'd makes drops visible again.  
The video you labeled as Vanilla looks to have Torch hit! installed, while the video labeled as Fabric looks to have both Torch hit! and Soul Fire'd installed.
Is everything correct?

Also, could you please send the debug.log from the server (both with and without mods)?

**Comment by OneDefauter** [comment]
> So, correct me if I'm wrong, but it seems to me that Torch hit! alone causes drops to be invisible, while Torch hit! + Soul Fire'd makes drops visible again.

No, the problem is not the 'Torch hit!', but the 'Soul Fire'd'. The 'Torch hit!' alone does what it has to do: set fire with the torch. But when the 'Soul Fire'd' is together on the server and the client doesn't have the mod the dropped items don't show up.


> The video you labeled as Vanilla looks to have Torch hit! installed, while the video labeled as Fabric looks to have both Torch hit! and Soul Fire'd installed.

In the video I labeled vanilla, it literally does not have a mod just the 'OptiFine' profile in the client. Both 'Torch hit!' and 'Soul Fire'd' are just installed on the server.
As for what I labeled 'Fabric', the client only has the 'Soul Fire'd' mod installed (and 'optifabric' and 'OptiFine'), the 'Torch hit!' is still only installed on the server.



> Also, could you please send the debug.log from the server (both with and without mods)?

I don't know where this 'debug.log' is, I only know 'latest.log'.

Client:
Vanilla (without mods, but with OptiFine profile)
1- (Client) [latest.log](https://github.com/Nyphet/soul-fire-d/files/10202824/latest.log)
Vanilla (without mods, without OptiFine profile)
2- Server with mod (Client): [latest.log](https://github.com/Nyphet/soul-fire-d/files/10202971/latest.log)


Vanilla (with mods)
3- (Client) [latest.log](https://github.com/Nyphet/soul-fire-d/files/10202829/latest.log)


Server:
With mods
4- (Server) [latest.log](https://github.com/Nyphet/soul-fire-d/files/10202826/latest.log)

Without mods
5- (Server) [latest.log](https://github.com/Nyphet/soul-fire-d/files/10202836/latest.log)
6- Cliente with mod (Client): [latest.log](https://github.com/Nyphet/soul-fire-d/files/10202837/latest.log)

These three 2, 5 and 6 'latest.log' caught my attention.

**Comment by Crystal-Spider** [comment]
So, sorry if I'm misunderstanding again.  
The problem happens when Soul Fire'd is installed on the Server but **not** on the Client, correct?

**Comment by OneDefauter** [comment]
> The problem happens when Soul Fire'd is installed on the Server but not on the Client, correct?

I thought the bug only happened if the server had the mod and the client did not, but in 'latest.log' 6 I did the test where the server was vanilla (no mods) while the client used the 'Soul Fire'd' mod and the result was that the items were not discarded.
I made a video to demonstrate
https://youtu.be/vLC6tc8cz-o

**Comment by Crystal-Spider** [comment]
Okay, now that I understand the situation I can tell you that this is not a bug.  
Soul Fire'd is required to be on **both** server **and** client, otherwise it won't work. If you look at the page on Curseforge or on GitHub it's stated clearly: 
![image](https://user-images.githubusercontent.com/65339023/207323698-70602628-8a4e-40b7-a8e3-4d2b594a83e0.png)  
Torch hit! is different because (unlike Soul Fire'd that is required on both sides) it is only required on the server, while it doesn't matter whether the client has it or not. This is also stated clearly on Torch hit! pages: 
![image](https://user-images.githubusercontent.com/65339023/207324079-1ebf15c6-e1e9-4a57-9afc-c1fff405d287.png)

So, to summarize, Torch hit! needs to be just server side, while Soul Fire'd needs to be on both client and server sides.  
Let me know if you need further assistance.
